src/mod_offline_mnesia.erl

```diff
@@ -162,3 +164,19 @@ remove_old_message(TimeStamp, Rec) ->
 
 is_old_message(MaxAllowedTimeStamp, #offline_msg{timestamp=TimeStamp}) ->
     TimeStamp < MaxAllowedTimeStamp.
+
+get_personal_data(Username, Server) ->
+    LUser = jid:nodeprep(Username),
+    LServer = jid:nodeprep(Server),
+    US = {LUser, LServer},
+    {atomic, Messages} = mnesia:transaction(fun() -> mnesia:wread({offline_msg, US}) end),
+    [{offline, ["timestamp", "from", "to", "packet"], offline_messages_to_gdpr_format(Messages)}].
+
+offline_messages_to_gdpr_format(MsgList) ->
+    [offline_msg_to_gdpr_format(Msg) || Msg <- MsgList].
+
+offline_msg_to_gdpr_format(#offline_msg{timestamp = Timestamp, from = From, to = To, packet = Packet}) ->
+    NowUniversal = calendar:now_to_universal_time(Timestamp),
+    {UTCTime, UTCDiff} = jlib:timestamp_to_iso(NowUniversal, utc),
+    UTC = list_to_binary(UTCTime ++ UTCDiff),
+    {UTC, jid:to_binary(jid:to_bare(From)), jid:to_binary(jid:to_bare(To)), exml:to_binary(Packet)}.
```

Comment:
Why are we converting these to bare JIDs?

Reply:
Ive pointed it in the PR's notes that Im returning bare jids.
I think I used bare jids because there was assertion in test for bare jid so I assumed that it is the requirement

Reply:
Yup, no need to convert them to bare. The test case was just a stub, written without examining what is exactly stored in the DB.